### ISSUE TITLE
Testing: Frontend

### ISSUE THREAD

**Issue by veitmo** [amosproj/amos2022ss04-digital-identity#33]
## User story
1. As a {HR employee}
2. I want / need {a well functioning frontend without bugs}
3. So that {I can efficiently and effectively interact with the system and trust in it}

## Acceptance criteria
* Create a general concept for testing of the frontend functionalities: How can a high quality of the components be assured?
* Think of first testing scenarios. Be able to present them, know broadly how they can be implemented. No need for actual implementation yet.

## SD criteria:
* Testing framework declared
* Documentation: how to install framework
* Test Template is available in git
* Example tests (1-3)

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by Idontker** [comment]
@veitmo 
I think the acceptance criteria need some work. It is a general task and the result is open: 
Do you want a template for testing future tasks? 
Maybe you want someone to create a small but specific test case in order to show a result. (As a SD you will have to implement a test case anyways to check whether the test framework works properly) 
The SDs will use the test cases to check automatically whether new features create bugs/errors.
For example, the release manager wants to have a set of tests he can run after merging features to check, whether old features are working fine.

I hope it helps to clarify why we (SD) want and need tests :)
